### ISSUE TITLE
[Bug] Twidere crashes when you attempt to follow someone and it runs into an error

### ISSUE THREAD

**Issue by solonovamax** [TwidereProject/TwidereX-Android#49]
## Issue description.
Today I just got sused and had to make a new account. Because of this, I had to follow ~750 accounts. Now, once I git about halfway through those accounts, twitter decided to follower limit me. Probably because it was going "dude what the fuck, you made your account 30 minutes ago, why are you following 350 people???"
So, I can't follow anyone because twitter just rejects it. Because I was having some issues with twitter being a pain in the ass and not letting me log into my new account, I decided to try twidereX again. And for shits and giggles, I was gonna try to follow someone.

Well, it turns out when you follow someone on twidere & are follower limited, it just crashes.

## Reproducing:

- Make a new twitter account
- add your email & phone so twitter doesn't suspend you automatically (bruh)
- Follow a shit ton of people until twitter limits you 
- Open twidere
- Try to follow someone
- ez crash

Here's an alternative version if you don't want to spend 2 hours copy & pasting URLs into your browser:
- Make a new twitter account
- do something to get suspended (basically, just don't register your phone, send a tweet or two, and then wait for like a few days for it to auto suspend you. I happen to have two accounts I don't give a shit about which have nothing on them & are suspended and I'd be willing to let you use them for testing.)
- Go to a user's profile & attempt to follow them
- ez crash

### Expected:
Twidere *shouldn't* crash when it runs into an error while following someone.
Or really anytime it's making a web request. For all the web requests, it should catch all the errors & display the stacktrace to the user along with a button to report it on github. (Opens an issue with the stack trace already in & a simple template. I'm like 99% sure that's possible.)

### Actual
Whenever it runs into an error it just d i e s

## What I think is happening
You forgot some catch blocks lol

(I wrote this on my phone, so if the formatting is dogshit, then I apologize for that. I'll fix it later.)

**Comment by Tlaster** [comment]
Yeah you're right, I forgot to catch the [follow](https://github.com/TwidereProject/TwidereX-Android/blob/develop/app/src/main/kotlin/com/twidere/twiderex/viewmodel/user/UserViewModel.kt#L84) request's error 😂

**Comment by solonovamax** [comment]
Yeah, you might also want to implement some kind of catch-all for exceptions like this, check the origin of the exception, and if it had something to do with a web request then it prevents a crash & displays a button to open a new issue.

Here's some code to generate a URL for an issue which contains the stack trace:
```kotlin
val repository = "TwidereProject/TwidereX-Android"
val title = "[Bug] ${e.javaClass.simpleName} when doing [...]"
val sw = StringWriter()
e.printStackTrace(PrintWriter(sw))
val stackTrace = sw.toString()
val body = """
|# Bug Report
|## Issue description.
|<!--Put a description of the issue here.-->
|
|## Reproducing:
|<!--Put the steps you must follow to trigger the bug.-->
|1. First step
|2. Second step
|3. ...
|
|## Expected:
|<!--Put what you expect to happen here-->
|
|## Actual
|<!--Put what is actually happening here-->
|
|## Stack trace
|```kotlin
|${stackTrace}```
|
|## Environment
|### Network type:
|<!--Place an X in the box for which network type you're using.-->
|- [ ] Cellular
|- [ ] Wi-Fi
|
|### Build Variant
|<!--Place an X in the box for which platform you're using.-->
|- [ ] Google Play
|- [ ] FDroid
|
|### App version:
|<!--Version Code here-->
""".trimMargin()
// if you don't like that huge block of text, here's a one liner:
// val body = "# Bug Report\n## Issue description.\n<!--Put a description of the issue here.-->\n\n## Reproducing:\n<!--Put the steps you must follow to trigger the bug.-->\n1. First step\n2. Second step\n3. ...\n\n## Expected:\n<!--Put what you expect to happen here-->\n\n## Actual\n<!--Put what is actually happening here-->\n\n## Stack trace\n```kotlin\n${stackTrace}```\n\n## Environment\n### Network type:\n<!--Place an X in the box for which network type you're using.-->\n- [ ] Cellular\n- [ ] Wi-Fi\n\n### Build Variant\n<!--Place an X in the box for which platform you're using.-->\n- [ ] Google Play\n- [ ] FDroid\n\n### App version:\n<!--Version Code here-->"
println(body)

val encodedTitle = URLEncoder.encode(title, "UTF-8")
val encodedBody = URLEncoder.encode(body, "UTF-8")
println(encodedTitle)
println(encodedBody)
val url = "https://github.com/${repository}/issues/new?title=${encodedTitle}&body=${encodedBody}"

// use the url somehow. Like put a button that opens this in a web browser.
// (Note: if it goes above ~2047 chars, web browsers shit themselves. Hopefully you use short names. By default it is 921 chars long.)
println(url)
println(url.length)
```

**Comment by Tlaster** [comment]
Fixed in 1.1.0
